lib/chef/win32/file.rb

```diff
@@ -60,6 +61,7 @@ def self.symlink(old_name, new_name)
         # TODO do a check for CreateSymbolicLinkW and
         # raise NotImplemented exception on older Windows
         flags = ::File.directory?(old_name) ? SYMBOLIC_LINK_FLAG_DIRECTORY : 0
+        flags |= SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE if Chef::ReservedNames::Win32::Version.new.win_10_creators_or_higher?
```

Comment:
Another approach here instead of checking for a specific windows version would be to check the error returned from `CreateSymbolicLinkW` for `ERROR_INVALID_PARAMETER` (87). Thats the [approach taken by rust](https://github.com/rust-lang/rust/blob/master/src/libstd/sys/windows/fs.rs#L649), but I think this is fine too.